src/MeasurementCodeInjector.php

```diff
@@ -70,58 +70,41 @@ public function injectEventTracking() {
         ?>
         <script>
             /**
-             * Keeps track of Ninja forms that have already been tracked
+             * Checks if the DOM event matches the Measurement Event or if any element in its path does
              *
-             * @type Array of elements
+             * @param e - DOM Event
+             * @param config - Measurement Event configuration
+             * @return {boolean}
              */
-            let ninjaFormsAddedNodes = [];
+            function pathHasMatch(e, config) {
```

Comment:
Let's check what performs better:

Get all targeted elements and check if there is a match vs. checking if the clicked target matches a given CSS rule.

https://developer.mozilla.org/en-US/docs/Web/API/Element/matches